src/microbe_stage/PlayerMicrobeInput.cs

```diff
@@ -313,7 +313,9 @@ public bool StartSprint()
             return false;
 
         ref var control = ref stage.Player.Get<MicrobeControl>();
-        control.Sprinting = true;
+
+        if (control.CanSprint)
```

Comment:
I would prefer an approach where this just sets Sprinting to true and relies on other systems to remove invalid sprint states. That's how the engulfing state works and makes logic much easier to write and less error-prone when calling code doesn't need to worry about if sprinting or engulfing is actually possible.